File: python/104-Maximum Depth of Binary Tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
  def DFS(self, curr):
    if curr == None:
      return 0
    left = self.DFS(curr.left)
    right = self.DFS(curr.right)
    return max(left, right) + 1
  def maxDepth(self, root):
    """
    :type root: Optional[TreeNode]
    :rtype: int
    """
    return self.DFS(root)

File: cpp/104-Maximum Depth of Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution
{
private:
  int DFS(const TreeNode* currNode)
  {
    if(nullptr == currNode)
    {
      return 0;
    }
    int left = DFS(currNode->left);
    int right = DFS(currNode->right);
    return std::max(left, right) + 1;
  }

public:
  int maxDepth(TreeNode* root)
  {
    int maxDepth = 0;
    maxDepth = DFS(root);
    return maxDepth;
  }
};